dotnet.ps1

```diff
@@ -0,0 +1,28 @@
+function Get-DotNetPath() {
+  $sdkCacheFile = (Join-Path $ToolsetDir 'sdk.txt')
+  Try {
```

Comment:
Would it be better to instead check whether `sdk.txt` exists to avoid the try/catch?

Reply:
the Try Catch here guards against concurrent write to the file. In case multiple processes want to write to that file at the same time. I could introduce a Mutex but I think that would be an overkill. I should add a comment.

Reply:
This does not fix the race condition reliably. If you are worried about race conditions like this, the best way to handle them is to create a temp file and then atomically rename the file.

Reply:
But doesn't the calling script `dotnet.cmd` need to know the location of that temp script as well? Maybe I'm misunderstanding something.

Reply:
I meant something like this:

```
$sdkCacheFileTemp = CommonLibrary\Get-TempPathFilename -Path  $ToolsetDir
Set-Content -Path $sdkCacheFileTemp -Value $dotnetPath

Try {
    Rename-Item -Path $sdkCacheFileTemp 'sdk.txt'
} Catch {
    # Somebody beat us
}
```

Reply:
Got it, thanks.